### PR TITLE
Add in a bunch of unit tests

### PR DESCRIPTION
**Pull request by alexlatchford**
Resolves [ZOMLINFRA-413](https://zbrt.atl.zillow.net/browse/ZOMLINFRA-413)

Done:
- Added tests for most major functionality.
- Added a bunch of useful fixtures for constructing temporary git repos (mostly taken from `pygit2.tests` and adapted) and on top of this constructing valid "template" repos.

### CODE REVIEW

File: battenberg/tmp_worktree.py [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+from pygit2 import Repository, Worktree
+from battenberg.errors import (
+    RepositoryEmptyException,
+    WorktreeConflictException,
+    WorktreeException
+)
+
+
+class TemporaryWorktree:
```
Comment (apryiomka) [inline battenberg/tmp_worktree.py]: Nit. do you want to rename the file to `temporary_worktree.py`?

Comment (apryiomka) [reply battenberg/tmp_worktree.py]: is this jut copy paste from the cut above?

Comment (alexlatchford) [reply battenberg/tmp_worktree.py]: Yep I just moved it into a separate module to make it easier to test it out. No changes. I can rename the module to `temporary_worktree.py`.

File: battenberg/temporary_worktree.py [hunk 2]
```diff
@@ -0,0 +1,60 @@
+import os
```
Comment (alexlatchford) [inline battenberg/temporary_worktree.py]: The changes in this file are just a copy operation to make it easier to test. This class used to live in `core.py` and moving it out to isolate it here.

**Pull request merged**